docs/user-guide/src/bmo/state_machine.md

```diff
@@ -0,0 +1,135 @@
+# Host State Machine
+
+During its lifetime, a `BareMetalHost` resource goes through a series of
+various states. Some of them are *stable* (the host stays in them indefinitely
+without user input), some are *transient* (the state will change once a certain
+operation completes). These fields in the `status` resource define the current
+state of the host:
+
+- `status.provisioning.state` -- the current phase of the provisioning process.
+- `status.operationHistory` -- the history of the main provisioning phases:
+  registration, inspection, provisioning and deprovisioning.
+- `status.operationalStatus` -- the overall status of the host.
+- `status.errorType` -- the type of the current error (if any).
+- `status.poweredOn` -- the current power state of the host.
+
+This is how the status of a healthy provisioned host may look like:
+
+```yaml
+status:
+ # ...
+ operationHistory:
+    deprovision:
+      end: null
+      start: null
+    inspect:
+      end: "2024-06-17T13:09:07Z"
+      start: "2024-06-17T13:03:54Z"
+    provision:
+      end: "2024-06-17T13:11:18Z"
+      start: "2024-06-17T13:09:26Z"
+    register:
+      end: "2024-06-17T13:03:54Z"
+      start: "2024-06-17T12:54:18Z"
+  operationalStatus: OK
+  poweredOn: true
+  provisioning:
+    ID: e09032ea-1b7d-4c50-bfcd-b94ff7e8d431
+    bootMode: UEFI
+    image:
+      checksumType: sha256
+      checksum: http://192.168.0.150/SHA256SUMS
+      format: qcow2
+      url: http://192.168.0.150/jammy-server-cloudimg-amd64.img
+    rootDeviceHints:
+      deviceName: /dev/sda
+    state: provisioned
+ # ...
+```
+
+## OperationalStatus
+
+- `OK` -- the host is healthy and operational.
+- `discovered` -- the host is known to Metal3 but lacks the required
+  information for the normal operation (usually, the BMC credentials).
+- `error` -- error has occured, see the `status.errorType` and
+  `status.errorMessage` fields for details.
+- `delayed` -- cannot proceed with the provisioning because the maximum number
+  of the hosts in the given state has been reached.
+- `detached` -- the host is detached, no provisioning actions are possible
+  (see [detached annotation](./detached_annotation.md) for details).
+
+## Provisioning state machine
+
+![BaremetalHost provisioning state transitions](images/provisioning-states.png)
```

Comment:
Is it worth hotlinking directly to the diagram in the repo (i.e. https://raw.githubusercontent.com/metal3-io/baremetal-operator/main/docs/BaremetalHost_ProvisioningState.png) so that we don't have to keep maintaining a copy over here?
It's hard enough to keep it in sync with the code when they live in the same repo.

Reply:
Yeah, I was thinking of dropping the diagram from there and redo this diagram to be more high-level.